extensions/espanso/src/addMatch.tsx

```diff
@@ -1,19 +1,50 @@
-import { Action, ActionPanel, Form, Toast, popToRoot, showToast } from "@raycast/api";
+import { Action, ActionPanel, Detail, Form, Toast, popToRoot, showToast } from "@raycast/api";
 import { FormValidation, useForm } from "@raycast/utils";
-import { homedir } from "os";
-import { join } from "path";
-import { appendMatchToFile, formatMatch, getAndSortTargetFiles } from "./utils";
+import { useEffect, useState } from "react";
 import { EspansoMatch } from "./types";
+import { appendMatchToFile, formatMatch, getAndSortTargetFiles, getEspansoConfig } from "./utils";
 
-const matchFilesDirectory: string = join(homedir(), "Library/Application Support/espanso/match");
-const sortedMatchFiles: { file: string; mtime: number }[] = getAndSortTargetFiles(matchFilesDirectory);
 export interface Values {
   trigger: string;
   replace: string;
   matchFile: string;
 }
 
 export default function Command() {
+  const [matchFilesDirectory, setMatchFilesDirectory] = useState<string>("null");
+  const [sortedMatchFiles, setSortedMatchFiles] = useState<{ file: string; mtime: number }[]>([]);
+  const [error, setError] = useState<Error | null>(null);
+
+  useEffect(() => {
+    async function getMatchFilesDirectory() {
```

Comment:
I prefer `const getMatchFilesDirectory = aynsc () =>` but it's just a preference.

Reply:
I wonder if you could export this function from the main command and pass functions as props.

Reply:
`async function getMatchFilesDirectory(matchFilesDirectory)`

Something like this. But maybe it's too much optimisation.

Reply:
I can try it out!

Reply:
Ended up going this direction:

```ts
  useEffect(() => {
    const fetchData = async () => {
      try {
        const { match: matchFilesDirectory } = await getEspansoConfig();
        const sortedMatchFiles: { file: string; mtime: number }[] = getAndSortTargetFiles(matchFilesDirectory);
        setMatchFilesDirectory(matchFilesDirectory);
        setSortedMatchFiles(sortedMatchFiles);
      } catch (err) {
        setError(err instanceof Error ? err : null);
      }
    };
    fetchData();
  }, []);
  ```
  Collapsing the two `useEffect()`s into one requiring only one call to `getEspansoConfig()`.